src/core-messages.lisp

```diff
@@ -59,46 +59,226 @@ relative to which control or readout waveforms may be defined.")
 
   :documentation "The resources required by a job")
 
-(defmessage |Waveform| ()
+(defmessage |AbstractWaveform| ()
```

Comment:
I think we should enforce the abstractness of this message class in the generated code, rather than making it a naming convention. That can be a follow-on PR if you'd rather not handle it here.

Reply:
P.S.: Open an issue about this if you don't want to solve it now.